openmls/src/treesync/mod.rs

```diff
@@ -68,6 +78,102 @@ pub use node::{leaf_node::LeafNode, parent_node::ParentNode, Node};
 #[cfg(any(feature = "test-utils", test))]
 pub mod tests_and_kats;
 
+/// An exported ratchet tree as used in, e.g., [`GroupInfo`](crate::messages::group_info::GroupInfo).
+#[derive(
+    PartialEq,
+    Eq,
+    Clone,
+    Debug,
+    Default,
+    Serialize,
+    Deserialize,
+    TlsSerialize,
+    TlsDeserialize,
+    TlsSize,
+)]
+// TODO(validation):
+//
+// * ValSemXXX: "If the tree has 2^d leaves, then it has 2^(d+1) - 1 nodes. The ratchet_tree vector logically has this number of entries, but the sender SHOULD NOT include blank nodes after the last non-blank node."
+// * ValSemXXX: "Regardless of how the client obtains the tree, the client MUST verify that the root hash of the ratchet tree matches the tree_hash of the GroupContext before using the tree for MLS operations."
+pub struct RatchetTreeExported(Vec<Option<Node>>);
```

Comment:
Why not generically call this `RatchetTree`? That way we can have a `RatchetTreeIn` in the context of validation.

Reply:
No good reason. Happy to rename it to `RatchetTree` when you don't see potential for confusion :+1:

Reply:
Ah... I renamed it to `RatchetTree` but could have already used `RatchetTreeIn`. Should we leave it that way anticipating this change or rename later? I'm leaning towards renaming later.